### PR TITLE
Attempt to fix issue #240

### PR DESCRIPTION
**Pull request by switch-cheehau**
#240

### CODE REVIEW

**Comment by switch-cheehau** [comment]
Hi Aimeos,

Please review the changes. thanks.

File: controller/common/src/Controller/Common/Order/Standard.php [hunk 1]
```diff
@@ -347,7 +348,13 @@ protected function updateStock( \Aimeos\MShop\Order\Item\Iface $orderItem, int $
...
 				foreach( $items as $item )
 				{
-					$stockManager->decrease( [$item->getProductId() => $how * -1 * $item->getQuantity()], $item->getStockType() );
+					if( $item->getType() === 'select' ) {
+						$productItem = $productManager->get( $item->getProductId(), array( 'product' ) );
+						$productRefItem = $productItem->getRefItems( 'product', 'default', 'default' )->where('product.code','=',$item->getProductCode())->first();
+						$stockManager->decrease( [$productRefItem->getId() => $how * -1 * $item->getQuantity()], $item->getStockType() );
+					} else {
+						$stockManager->decrease( [$item->getProductId() => $how * -1 * $item->getQuantity()], $item->getStockType() );
+					}
```
Comment (aimeos) [inline controller/common/src/Controller/Common/Order/Standard.php:357]: Try this:
```php
$prodId = $item->getProductId();

if( $item->getType() === 'select' )
{
	$prodId = $productManager->get( $item->getProductId(), ['product' => ['default']] )
		->getRefItems( 'product', null, 'default' )
		->where( 'product.code', '==', $item->getProductCode() )
		->getId()->first();
}

$stockManager->decrease( [$prodId => $how * -1 * $item->getQuantity()], $item->getStockType() );
```
This fixes a few potential problems in your code.

Comment (switch-cheehau) [reply controller/common/src/Controller/Common/Order/Standard.php]: updated the code, tested manually, it works. Thanks for the guide.

how do i adapt the test? 

image

File: controller/common/src/Controller/Common/Order/Standard.php [hunk 2]
```diff
@@ -347,7 +348,16 @@ protected function updateStock( \Aimeos\MShop\Order\Item\Iface $orderItem, int $
...
 				{
-					$stockManager->decrease( [$item->getProductId() => $how * -1 * $item->getQuantity()], $item->getStockType() );
+					$prodId = $item->getProductId();
+
+					if( $item->getType() === 'select' ) {
+						$prodId = $productManager->get( $item->getProductId(), ['product' => ['default']] )
+							->getRefItems( 'product', null, 'default' )
+							->where( 'product.code', '==', $item->getProductCode() )
+							->getId()->first();
+					}
```
Comment (aimeos) [inline controller/common/src/Controller/Common/Order/Standard.php:358]: There's already a switch statement below with calls `updateStockSelection()`. If you move your code to that method, testing may be easier.

**Comment by aimeos** [comment]
Fixed in https://github.com/aimeos/aimeos-core/commit/e2b1191bdcd74e0e9b0fb51de978de121b852d5e

**Pull request closed without merging**